docs/reference/modules/network/threading.asciidoc

```diff
@@ -0,0 +1,106 @@
+Similarly, outgoing data are not actually sent over a channel until the owning
+`transport_worker` thread is free. This means that we require every
+`transport_worker` thread to be idle frequently, which looks something like
+this in a stack dump:
+
+```
+"elasticsearch[instance-0000000000][transport_worker][T#1]" #28 daemon prio=5 os_prio=0 cpu=1232378.30ms elapsed=47152.02s tid=0x00007fc8a8569da0 nid=0x16f runnable  [0x00007fc8ac5e6000]
+   java.lang.Thread.State: RUNNABLE
+        at sun.nio.ch.EPoll.wait(java.base@15.0.1/Native Method)
+        at sun.nio.ch.EPollSelectorImpl.doSelect(java.base@15.0.1/EPollSelectorImpl.java:120)
+        at sun.nio.ch.SelectorImpl.lockAndDoSelect(java.base@15.0.1/SelectorImpl.java:129)
+        - locked <0x0000000624849c60> (a sun.nio.ch.Util$2)
+        - locked <0x0000000624849c08> (a sun.nio.ch.EPollSelectorImpl)
+        at sun.nio.ch.SelectorImpl.select(java.base@15.0.1/SelectorImpl.java:146)
+        at io.netty.channel.nio.NioEventLoop.select(NioEventLoop.java:803)
+        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:457)
+        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
+        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
+        at java.lang.Thread.run(java.base@15.0.1/Thread.java:832)
+```
+
+In the <<cluster-nodes-hot-threads>> API an idle `transport_worker` thread will
+be reported like this:
+
+```
+100.0% [cpu=0.0%, other=100.0%] (500ms out of 500ms) cpu usage by thread 'elasticsearch[instance-0000000000][transport_worker][T#1]'
+     10/10 snapshots sharing following 9 elements
+       java.base@17.0.1/sun.nio.ch.EPoll.wait(Native Method)
+       java.base@17.0.1/sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:118)
+       java.base@17.0.1/sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:129)
+       java.base@17.0.1/sun.nio.ch.SelectorImpl.select(SelectorImpl.java:146)
+       io.netty.channel.nio.NioEventLoop.select(NioEventLoop.java:810)
+       io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:457)
+       io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
+       io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
+       java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
+```
+
+Note that `transport_worker` threads should always be in state `RUNNABLE`,
+because they block in the native `EPoll#wait` method, so the hot threads API
+will report them at 100% overall utilisation. This is normal, and the breakdown
+of time into `cpu=` and `other=` fractions shows how much time the thread spent
+running and waiting for input respectively.
+
+If a `transport_worker` thread is not frequently idle then it will build up a
+backlog of work, which will cause delays in processing messages on the channels
+that it owns. It is hard to predict exactly which work will be delayed:
```

Comment:
```suggestion
If a `transport_worker` thread is not frequently idle, it will build up a
backlog of work. This can cause delays in processing messages on the channels
that it owns. It's hard to predict exactly which work will be delayed:
```